armi/reactor/converters/uniformMesh.py

```diff
@@ -100,14 +105,26 @@ def _checkConversion(self):
 
     @staticmethod
     def initNewReactor(sourceReactor):
-        """Build an empty version of the new reactor"""
-        # XXX: this deepcopy is extremely wasteful because the assemblies copied
-        # are immediately removed. It's just laziness of getting the same class
-        # of reactor set up.
-        newReactor = copy.deepcopy(sourceReactor)
-        newReactor.core.removeAllAssemblies()
-        newReactor.core.regenAssemblyLists()
-        return newReactor
+        """Build a new, yet empty, reactor with the same settings as sourceReactor
+
+        Parameters
+        ----------
+        sourceReactor : :py:class:`Reactor <armi.reactor.reactors.Reactor>` object.
+            original reactor to be copied
+        """
+        # create new operator
+        newOper = operators.factory(sourceReactor.o.cs)
+        # create new reactor from blueprints used to create sourceReactor
+        bp = blueprints.loadFromCs(sourceReactor.o.cs)
```

Comment:
# create new reactor from blueprints used to create sourceReactor. Additionally,
        # the attributes on the original reactor's blueprints should be copied over.
        bp = blueprints.loadFromCs(sourceReactor.o.cs)
        bp.allNuclidesInProblem = sourceReactor.blueprints.allNuclidesInProblem
        bp.activeNuclides = sourceReactor.blueprints.activeNuclides
        bp.inertNuclides = sourceReactor.blueprints.inertNuclides
        bp.elementsToExpand = sourceReactor.blueprints.elementsToExpand